patches/tModLoader/Terraria/Player_tML.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using Terraria.ModLoader;
+
+namespace Terraria
+{
+	public partial class Player
+	{
+		private DamageClassData[] damageData = new DamageClassData[0];
+
+		internal void ResetDamageClassDictionaries() {
+			damageData = new DamageClassData[DamageClassLoader.DamageClassCount];
+			for (int i = 0; i < DamageClassLoader.DamageClassCount; i++) {
+				damageData[i] = new DamageClassData(0, 0, 1);
+			}
+		}
+
+		/// <summary>
+		/// Edits the crit for the given damage type on this player.
+		/// </summary>
+		/// <param name="changeAmount">The amount that should be added to the crit. Can be negative too.</param>
+		public void SetCrit<T>(int changeAmount) where T : ModDamageClass {
+			int damageClassType = DamageClassLoader.GetIndex<T>();
+			if (ContainsIndex(damageData, damageClassType)) {
+				ref var data = ref damageData[damageClassType];
+				data.crit += changeAmount;
+			}
+		}
+
+		/// <summary>
+		/// Edits the damage value for the given damage type on this player.
+		/// </summary>
+		/// <param name="changeAmount">The amount (as a percentage, 0.01f is 1%) that should be added to the damage. Can be negative too.</param>
+		public void SetDamage<T>(float changeAmount) where T : ModDamageClass {
+			int damageClassType = DamageClassLoader.GetIndex<T>();
+			if (ContainsIndex(damageData, damageClassType)) {
+				ref var data = ref damageData[damageClassType];
+				data.damage += changeAmount;
+			}
+		}
+
+		/// <summary>
+		/// Edits the damage multiplier's value for the given damage type on this player.
+		/// </summary>
+		/// <param name="changeAmount">The amount (as a percentage, 0.01f is 1%) that should be added to the damage multiplier. Can be negative too.</param>
+		public void SetDamageMult<T>(float changeAmount) where T : ModDamageClass {
```

Comment:
Why do all these have 'Set' in the name, if they add to values instead of setting them?

Reply:
Renamed to 'Modify' for clarity